environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentCreationService.java

```diff
@@ -65,14 +82,49 @@ public EnvironmentDto create(EnvironmentCreationDto creationDto, String accountI
         environmentService.setSecurityAccess(environment, creationDto.getSecurityAccess());
         CloudRegions cloudRegions = setLocationAndRegions(creationDto, environment);
         validateCreation(creationDto, environment, cloudRegions);
+        Map<String, CloudSubnet> subnetMetas = networkService.retrieveSubnetMetadata(environment, creationDto.getNetwork());
+        validateNetworkRequest(environment, creationDto.getNetwork(), subnetMetas);
         environment = environmentService.save(environment);
-        environmentResourceService.createAndSetNetwork(environment, creationDto.getNetwork(), accountId);
+        environmentResourceService.createAndSetNetwork(environment, creationDto.getNetwork(), accountId, subnetMetas);
         createAndSetParameters(environment, creationDto.getParameters(), accountId);
         environmentService.save(environment);
         reactorFlowManager.triggerCreationFlow(environment.getId(), environment.getName(), creator, environment.getResourceCrn());
         return environmentDtoConverter.environmentToDto(environment);
     }
 
+    private void validateNetworkRequest(Environment environment, NetworkDto network, Map<String, CloudSubnet> subnetMetas) {
+        String message;
+        if (AWS.name().equalsIgnoreCase(environment.getCloudPlatform()) && network != null && network.getNetworkCidr() == null) {
+            if (network.getSubnetIds().size() < 2) {
+                message = "Cannot create environment, there should be at least two subnets in the network";
+                LOGGER.info(message);
+                throw new BadRequestException(message);
```

Comment:
These messages are thrown in their own separate exception respectfully, however these errors are not exclusive. More of these validation errors could occur in the same request but the user will only by notified about the first error. 

Please use the `com.sequenceiq.cloudbreak.util.ValidationResult` class to store multiple validation errors and notify the user about all of the errors.

Reply:
ok